frontend/src/app/map/map.component.ts

```diff
@@ -27,88 +41,244 @@ import { NotificationService } from '../notification.service';
+
+  polygonCounter = 1;
+  radiusCounter = 1;
+
+  addressIsLoading = false;
+
+  public DrawObjectType = DrawObjectType;
+  drawType = DrawObjectType.CIRCLE
+
+  searchAreas: DisplayFeatures[] = [];
+
   constructor(
     private coordinateService: CoordinateService,
     private apiService: ApiService,
     private translate: TranslateService,
     private notificationService: NotificationService
-    ) {}
+  ) { }
 
   ngOnInit() {
     this.initializeMap();
-    this.initializeMarkerLayer();
-  }
 
-  ngAfterViewInit() {
-    this.addClickListener();
+    if (!this.enableDrawFeatures) {
+      this.addClickListener();
+    } else {
+      this.addSubscription();
+      this.addInteraction();
+    }
+
+    this.drawPoints();
   }
 
-  /** Initializes the OpenLayers map with a tile layer and default view*/
+
   initializeMap() {
-    const rasterLayer = new TileLayer({
+    this.source = new VectorSource({ wrapX: false });
+    this.vector = new VectorLayer({
+      source: this.source,
+    });
+
+    this.popupSource = new VectorSource({})
+    this.popupLayer = new VectorLayer({
+      source: this.popupSource,
+      style: new Style({
+        image: new Circle({
+          radius: 5,
+          fill: new Fill({
+            color: '#FF0000'
+          })
+        })
+      })
+    });
+
+    this.pointSource = new VectorSource({})
+    this.pointLayer = new VectorLayer({
+      source: this.pointSource,
+      style: new Style({
+        image: new Circle({
+          radius: 4.5,
+          fill: new Fill({
+            color: '#F54FA6'
+          }),
+          stroke: new Stroke({
+            color: '#FFFFFF',
+            width: 2
+          })
+        })
+      })
+    });
+
+    const raster = new TileLayer({
       source: new XYZ({
         url: 'https://{a-c}.tile.openstreetmap.org/{z}/{x}/{y}.png'
       })
     });
 
     this.map = new Map({
       target: 'map',
-      layers: [rasterLayer],
+      layers: [raster, this.vector, this.popupLayer, this.pointLayer],
       view: new View({
-        center: fromLonLat([13.404954, 52.520008]),
-        zoom: 12
+        center: fromLonLat([13.404954, 52.520008]), // Berlin coordinates
+        zoom: 12,
       })
     });
+
+    this.modify = new Modify({ source: this.source });
+    this.map.addInteraction(this.modify);
+    this.map.addInteraction(new Snap({ source: this.source }));    
   }
 
-  /**
-   * Initializes the vector layer for displaying markers on the map
-   * https://openlayers.org/en/latest/apidoc/module-ol_layer_Vector-VectorLayer.html 
-   */
-  initializeMarkerLayer() {
-    this.vectorSource = new VectorSource();
-    this.vectorLayer = new VectorLayer({
-      source: this.vectorSource,
-      style: new Style({
-        image: new Circle({
-          radius: 5,
-          fill: new Fill({
-            color: '#FF0000'
-          })
-        })
+  drawPoints(){
+    if(this.coordinatesToDisplay){
+      this.coordinatesToDisplay.forEach(coords => {
```

Comment:
could also be written as
```ts
for(let coords of this.coordinatesToDisplay || []) { 
     ...
}
```
thus no if statement and one level of code depth gets eliminated. But this is not critical ^^ just wanted to share